### PR TITLE
feat: Add bill coupon

### PR DESCRIPTION
**Pull request by Naman-Saxena1**
# Added Coupon on order bill
1. User can use Coupon BOOKS200 to get 200 off 

# Added features that I previously forgot to add as per PRD
1. In Wishlist - Add to Cart button
2. In Cart - Add to Wishlist button

# Files to review :
1. ShoppingBill.css
2. ShoppingBill,jsx

# Coupon Feature Screenshot
Bookztron-CartPage-1

### CODE REVIEW

File: src/Components/HorizontalCard/HorizontalProductCard.jsx [hunk 1]
```diff
@@ -96,6 +98,48 @@ function HorizontalProductCard({productDetails})
...
+        if(token)
+        {
+            const user = jwt_decode(token)
+            
+            if(!user)
+            {
+                localStorage.removeItem('token')
+                showToast("warning","","Kindly Login")
+                navigate('/login')
+            }
```
Comment (bhtibrewal) [inline src/Components/HorizontalCard/HorizontalProductCard.jsx:114]: can't understand why r u doing this

File: src/Components/HorizontalCard/HorizontalProductCard.jsx [hunk 2]
```diff
@@ -96,6 +98,48 @@ function HorizontalProductCard({productDetails})
...
+            else
+            {
+                let wishlistUpdateResponse = await axios.patch(
```
Comment (bhtibrewal) [inline src/Components/HorizontalCard/HorizontalProductCard.jsx]: why is it let and not const

File: src/Components/HorizontalCard/HorizontalProductCard.jsx [hunk 3]
```diff
@@ -96,6 +98,48 @@ function HorizontalProductCard({productDetails})
         } 
     }
 
+    async function addItemToWishlist()
```
Comment (bhtibrewal) [inline src/Components/HorizontalCard/HorizontalProductCard.jsx:101]: Suggestion: move it to a separate file

File: src/Components/ShoppingBill/ShoppingBill.jsx [hunk 4]
```diff
@@ -1,18 +1,29 @@
...
+        finalBill = totalBill - 200;
+    }
+    else
+    {
+        finalBill = totalBill;
+    }
+
     function placeOrder()
     {
         showToast("info","","Payment Integration is coming soon! 😉")
```
Comment (bhtibrewal) [inline src/Components/ShoppingBill/ShoppingBill.jsx:29]: Liked this thought

**Review by bhtibrewal — APPROVED** [review]
The code is good clean just some small suggestions, and you are good to merge

**Review by Sahil25061999 — APPROVED** [review]
No major changes required, the code looks good, all the conventions have been followed, liked the naming conventions. just a suggestion after signing up it will good if the user is directly redirected to the homepage instead of login in

File: src/Components/HorizontalCard/HorizontalProductCard.jsx (hunk 1 above)
Comment (Naman-Saxena1) [reply src/Components/HorizontalCard/HorizontalProductCard.jsx:114]: It is to check if jwt payload is present after decoding

File: src/Components/HorizontalCard/HorizontalProductCard.jsx (hunk 2 above)
Comment (Naman-Saxena1) [reply src/Components/HorizontalCard/HorizontalProductCard.jsx]: Change Done

File: src/Components/WishlistProductCard/WishlistProductCard.jsx [hunk 5]
```diff
@@ -0,0 +1,237 @@
...
+        imgSrc, 
+        imgAlt,
+        badgeText, 
+        outOfStock
+    } = productdetails
+    const [wishlistHeartIcon, setWishlistHeartIcon] = useState("fa-heart-o")
+    const [wishlistBtn, setWishlistBtn]             = useState("add-to-wishlist-btn")
+
+    useEffect(()=>{
+        const index = userWishlist.findIndex(product=> {
```
Comment (gouravmarch20) [inline src/Components/WishlistProductCard/WishlistProductCard.jsx:35]: avoid nested if else  repalce it by  ==< switch case or ternary operator  or if block only required 

in else all had included all the case of failure , but be specific what kind of error you are getting and according to write logic

File: src/Components/WishlistProductCard/WishlistProductCard.jsx [hunk 6]
```diff
@@ -0,0 +1,237 @@
...
+                
+                if(!user)
+                {
+                    localStorage.removeItem('token')
+                    showToast("warning","","Kindly Login")
+                    navigate('/login')
+                }
+                else
+                {
+                    let wishlistUpdateResponse = await axios.delete(
```
Comment (gouravmarch20) [inline src/Components/WishlistProductCard/WishlistProductCard.jsx:114]: create a sperate folder , to hit all backend request 

it help to sport backend response & return response to all component when ever it needed

**Review by gouravmarch20 — COMMENTED** [review]
can be fix :

- seperate folder to hit all axios request 
- avoid too  much else because you have to be specific

**Pull request merged**